cmd/mkroot/commands.go

```diff
@@ -174,125 +148,141 @@ func useLocalInit(repo string) (initPath string, runConfig []byte, err error) {
 	return
 }
 
-func (m *mkfs) Execute() error {
+func (m *mkfs) createLoopDrive() error {
 	fallocate := exec.Command(fallocatePath, "-l", m.size, m.name)
 	if err := fallocate.Run(); err != nil {
-		return nil
+		return fmt.Errorf("failed to allocate file '%s', %w", m.name, err)
 	}
 
-	if !Exists(m.fscmd) {
-		return fmt.Errorf("could not find '%s'", m.fscmd)
-	}
 	mkfs := exec.Command(m.fscmd, m.name)
 	if err := mkfs.Run(); err != nil {
 		return fmt.Errorf("failed to create filesystem: %v", err)
 	}
+	return nil
+}
 
-	if m.fromImage {
-		temp, cleanFunc, err := mktemp()
-		if err != nil {
-			return fmt.Errorf("error creating tmp directory: %w", err)
-		}
-
-		defer func() {
-			if err := cleanFunc(); err != nil {
-				log.Fatal("error cleaning up resources: %w", err)
-			}
-		}()
+func (m *mkfs) createInitDrive() error {
```

Comment:
this one creates a loop drive containing our init

Reply:
Wouldn't it be a better idea to just have a `createDrive` function that creates the virtual drive and formats the filesystem unto it? I think this would work better because creating the init and rootfs drives all need to create drives.